src/plugins/intel_cpu/src/nodes/executors/kleidiai/kleidiai_mm.cpp

```diff
@@ -0,0 +1,159 @@
+        aclfcAttrs.isConvertedWeights = true;
+    }
+    auto originalWeightsDesc = memory.at(ARG_WEI)->getDescPtr();
+    const auto& wgtDims = originalWeightsDesc->getShape().getStaticDims();
+    const VectorDims wgtDims2D = reshapeDownToRank<2>(wgtDims);
+    originalWeightsDesc = std::make_shared<CpuBlockedMemoryDesc>(originalWeightsDesc->getPrecision(), Shape{wgtDims2D});
+    auto dnnlSrcDesc = MemoryDescUtils::convertToDnnlMemoryDesc(originalWeightsDesc);
+    auto dstDesc = originalWeightsDesc->cloneWithNewPrecision(memory.at(ARG_SRC)->getDescPtr()->getPrecision());
+    auto dnnlDstDesc = MemoryDescUtils::convertToDnnlMemoryDesc(dstDesc);
+    if (!attrs.weightsNonTransposed) {
+        dnnlDstDesc = acl_fc_executor::makeTransposedWeightDescriptor(dnnlDstDesc, dnnlSrcDesc);
+        aclfcAttrs.isWeightsRepacked = true;
+    }
+    packedWeights = acl_fc_executor::reorderWeights(memory, context, aclfcAttrs, dnnlSrcDesc, dnnlDstDesc);
+}
+
+bool MatMulKleidiAIExecutor::update(const MemoryArgs& memory) {
+    const auto& weiDesc = memory.at(ARG_WEI)->getDescPtr();
+    const auto& dstDesc = memory.at(ARG_DST)->getDescPtr();
+    const auto& wgtDims = weiDesc->getShape().getStaticDims();
+    // Weights are transposed by MatMulConstTransposesExtraction
+    // K is the IC of weight
+    // the weight is reshaped to [-1, K] in ConvertMatMulToFC
+    K = wgtDims[1];
+    N = wgtDims[0];
+
+    const auto& outDims = dstDesc->getShape().getStaticDims();
+    if (outDims.size() > 2) {
+        M = std::accumulate(outDims.begin(), outDims.end() - 1, 1, std::multiplies<size_t>());
+    } else {
+        M = outDims[0];
+    }
+    return true;
+}
+
+void MatMulKleidiAIExecutor::execute(const MemoryArgs& memory) {
+    size_t BLOCK_SIZE = 8;
+    auto srcMem = memory.at(ARG_SRC);
+    auto weiMem = memory.at(ARG_WEI);
+    auto dstMem = memory.at(ARG_DST);
+    auto biasMem = memory.at(ARG_BIAS);
+    auto srcDims = normalizeDimsTo2D(srcMem->getDesc().getShape().getDims());
+    auto weiDims = weiMem->getDesc().getShape().getDims();
+    auto M = srcDims[0];
+    auto K = srcDims[1];
+    auto N = weiDims[0];
+
+    const size_t lhs_stride = K * sizeof(float);
+    const size_t rhs_stride = N * sizeof(float);
+    const size_t dst_stride_row = N * sizeof(float);
+    const size_t dst_stride_col = sizeof(float);
+
+    const size_t rhs_packed_size = kai_get_rhs_packed_size_rhs_pack_kxn_f32p8x1biasf32_f32_f32_neon(N, K);
+    float* rhs_packed = new float[rhs_packed_size];
+
+    const size_t nr = ukernel.get_nr();
+    const size_t kr = ukernel.get_kr();
+    const size_t sr = ukernel.get_sr();
+
+    float* lhs = srcMem->getDataAs<float>();
+    float* rhs = static_cast<float*>(packedWeights->getData());
+    float* dst = dstMem->getDataAs<float>();
+    float* bias = biasMem->getDataAs<float>();
+
+    if (bias == nullptr) {
+        bias = new float[N];
+        memset(bias, 0, N * sizeof(float));
+    }
```

Comment:
1. Memory allocations for packed weights and bias shouldn't be done on each inference, Lets move to MatMulKleidiAIExecutor ctor instead.
2. Memory is allocated, but not deallocated. Lets use ov::intel_cpu:;Memory as data object instead of direct `new` calls

Reply:
1. I moved bias allocation to constructor, however packed weights are allocated in ctor already in `acl_fc_executor::reorderWeights` method.
2. Fixed